src/LevelZeroAcceleratorTopo.cpp

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2015 - 2021, Intel Corporation
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in
+ *       the documentation and/or other materials provided with the
+ *       distribution.
+ *
+ *     * Neither the name of Intel Corporation nor the names of its
+ *       contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <iostream>
+#include <string>
+#include <map>
+
+#include "config.h"
+#include "Exception.hpp"
+#include "LevelZeroDevicePool.hpp"
+#include "LevelZeroAcceleratorTopo.hpp"
+
+namespace geopm
+{
+    LevelZeroAcceleratorTopo::LevelZeroAcceleratorTopo()
+        : LevelZeroAcceleratorTopo(levelzero_device_pool(), geopm_sched_num_cpu())
+    {
+    }
+
+    LevelZeroAcceleratorTopo::LevelZeroAcceleratorTopo(const LevelZeroDevicePool &device_pool,
+                                                       const int num_cpu)
+        : m_levelzero_device_pool(device_pool)
+    {
+        unsigned int num_accelerator = m_levelzero_device_pool.
+                                       num_accelerator(GEOPM_DOMAIN_BOARD_ACCELERATOR);
+        unsigned int num_accelerator_chip = m_levelzero_device_pool.
+                                            num_accelerator(GEOPM_DOMAIN_BOARD_ACCELERATOR_CHIP);
+
+        if (num_accelerator == 0) {
+            std::cerr << "Warning: <geopm> LevelZeroAcceleratorTopo: No levelZero devices detected.\n";
```

Comment:
should this be in a GEOPM_DEBUG clause, or be a throw?

Reply:
DEBUG.